bot/utils/blurple_formatter.py

```diff
@@ -0,0 +1,717 @@
+        op_type = type(op)
+        if isinstance(left, (ast.BoolOp, ast.BinOp)):
+            left_type = type(left.op)
+        if isinstance(right, (ast.BoolOp, ast.BinOp)):
+            right_type = type(right.op)
+
+        par_left = (
+            precedences[left_type] < precedences[op_type]
+            if op_type not in r_assoc
+            else precedences[left_type] <= precedences[op_type]
+        )
+        par_right = (
+            precedences[right_type] <= precedences[op_type]
+            if op_type not in r_assoc
+            else precedences[right_type] < precedences[op_type]
+        )
+        nl_before = (op_type in nl_before_op) * "\n"
+        nl_after = (op_type not in nl_before_op) * "\n"
+        if nl_able:
+            return (
+                f"{(parenthesize if par_left else unparse)(left, nl_able)}{space()}"
+                + f"{nl_before}{op_strs[type(op)]}{space()}{nl_after}"
+                + f"{(parenthesize if par_right else unparse)(right, nl_able)}"
+            )
+        else:
+            return (
+                f"({(parenthesize if par_left else unparse)(left, True)}{space()}"
+                + f"{nl_before}{op_strs[type(op)]}{space()}{nl_after}"
+                + f"{(parenthesize if par_right else unparse)(right, True)})"
+            )
+    if isinstance(node, ast.Call):
+        func, args, keywords = node.func, node.args, node.keywords
+
+        return (
+            f"{(parenthesize if precedences[type(func)] < precedences[ast.Call] else unparse)(func)}{space()}"
+            f"({space()}{f',{space()}'.join(unparse(arg, True) for arg in args)}"
+            + (
+                (f",{space()}" if args else "")
+                + f",{space()}".join(unparse(kwarg, True) for kwarg in keywords)
+                if keywords
+                else ""
+            )
+            + space()
+            + ")"
+        )
+    if isinstance(node, ast.ClassDef):
+        name, bases, keywords, body, decorator_list = (
+            node.name,
+            node.bases,
+            node.keywords,
+            node.body,
+            node.decorator_list,
+        )
+
+        args = bases + keywords
+        return "\n".join(
+            [
+                *(f"@{space()}{unparse(decorator)}" for decorator in decorator_list),
+                f"class{space(1)}{name}{space()}"
+                + (
+                    f"({space()}{f',{space()}'.join(unparse(arg, True) for arg in args)}{space()})"
+                    if args
+                    else ""
+                )
+                + f"{space()}:",
+                *indent(body, randrange(2, 5)),
+            ]
+        )
+    if isinstance(node, ast.Compare):
+        left, ops, comparators = node.left, node.ops, node.comparators
+
+        return (
+            f"{(parenthesize if precedences[type(left)] <= precedences[ast.Compare] else unparse)(left)}"
+            + f"{space()}".join(
+                f"{space()}{op_strs[type(op)]}{space()}"
```

Comment:
Isn't there already space in front of it?
```suggestion
                f"{op_strs[type(op)]}{space()}"
```

Reply:
There is a space in front for the first iteration of the loop, but not the next iterations